analyzers/src/SonarAnalyzer.CSharp/Rules/StringLiteralShouldNotBeDuplicated.cs

```diff
@@ -60,5 +71,8 @@ public sealed class StringLiteralShouldNotBeDuplicated : StringLiteralShouldNotB
 
         protected override string GetLiteralValue(LiteralExpressionSyntax literal) =>
             literal.Token.Text;
+
+        protected override bool IsNotNamedTypeOrTopLevelMain(SyntaxNodeAnalysisContext context) =>
```

Comment:
I find the negation in the method name very hard to read. It's also a bit confusing since I don't know from start if the negation applies for top-level main too.

I suggest to re-write the code to avoid the negation in the method name.
Additionally, we can split this in 2 different methods: `IsTopLevelMain` and `IsNamedType`